src/tiled-map-resource.ts

```diff
@@ -440,7 +466,7 @@ export class TiledMapResource implements Loadable<TiledMap> {
             const tileMapLayer = new TileMap(0, 0, this.data.rawMap.tilewidth, this.data.rawMap.tileheight, this.data.height, this.data.width);
             tileMapLayer.addComponent(new TiledLayerComponent(layer));
 
-            const zindex = getProperty<number>(rawLayer.properties, 'zindex')?.value || layerZIndexBase++;
+            const zindex = getProperty<number>(rawLayer.properties, 'zindex')?.value ?? (layer.rawLayer.order + layerZIndexBase);
```

Comment:
This is the 3rd or 4th time I've seen this code, candidate to refactor and extract into helper. Note this one also does not coerce when setting 😊

Reply:
Good call, that's a good refactoring